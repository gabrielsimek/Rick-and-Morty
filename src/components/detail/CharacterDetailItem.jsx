import React from 'react';
import PropTypes from 'prop-types';

const CharacterDetailItem = (
  { character, name, species, status, image, onClick }) => {
  //refactor this
  return (
    <figure>
      <img 
        src={character.image}/>
      <figcaption>
        <h2>{character.name}</h2>
        <p>{character.species}</p>
        <p>{character.status}</p>
      </figcaption>
      <button
        onClick={onClick}
      >Back</button>
    </figure>
  );
};

CharacterDetailItem.propTypes = {
  // name: PropTypes.string.isRequired,
  // species: PropTypes.string.isRequired,
  // status: PropTypes.string.isRequired,
  // image: PropTypes.string.isRequired,
  character: PropTypes.object.isRequired,
  onClick: PropTypes.func.isRequired

};

export default CharacterDetailItem;
