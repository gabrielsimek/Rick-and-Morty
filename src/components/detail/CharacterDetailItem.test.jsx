import React from 'react';
import { render, cleanup } from '@testing-library/react';
import CharacterDetailItem from './CharacterDetailItem';
import { MemoryRouter } from 'react-router-dom';
describe('App CharacterDetailItem', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<MemoryRouter>
      <CharacterDetailItem character={
        { name: 'name', species: 'species', status: 'status', image: 'image' }
      }/></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
  });
});
