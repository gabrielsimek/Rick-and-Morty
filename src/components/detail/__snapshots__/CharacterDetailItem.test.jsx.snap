// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App CharacterDetailItem renders App 1`] = `
<DocumentFragment>
  <figure>
    <img
      src="image"
    />
    <figcaption>
      <h2>
        name
      </h2>
      <p>
        species
      </p>
      <p>
        status
      </p>
    </figcaption>
    <button>
      Back
    </button>
  </figure>
</DocumentFragment>
`;
